flow/raster_cache.h

```diff
@@ -88,6 +92,31 @@ struct RasterCacheMetrics {
 
 class RasterCache {
  public:
+  struct Context {
+    GrDirectContext* gr_context;
+    SkColorSpace* dst_color_space;
+    const SkMatrix& matrix;
+    const SkRect& logical_rect;
+    const char* flow_type;
+    bool checkerboard;
+  };
```

Comment:
At first I thought this was a back-step from what I proposed because leaving out the matrix, rect and flow_type fields allowed the structure to be const for the entire processing of all of the cache entries. It does make the method signature in RasterCache simpler, though. My concern was that constructing a context object per usage takes time and simply replaces arguments to the method.

One thing I noticed. "flow_type" is not an optional parameter here, yet you construct 3 of these and none of them fill in a value for it. When it was a parameter to the Update method, it was hard to miss providing a value. If they don't get filled in then we might get crashes from the code that creates flow events.